graph/patterns/longRunningOperations.md

```diff
@@ -0,0 +1,192 @@
+    SHOULD be able to determine the system state even if the operation tracking
+    resource is no longer active. Clients MAY issue a GET on some resource to
+    determine the state of a long running operation.
+
+-  The long running operations pattern SHOULD work for clients looking to "fire and forget"
+    and for clients looking to actively monitor and act upon results.
+
+-  The long running operations pattern might be supplemented by the [change notification
+    pattern](change-notification.md).
+
+-  Cancellation of a long running operation does not explicitly mean a rollback. On a per API-defined case, it
+    might mean a rollback, or compensation, or completion, or partial completion,
+    etc. Following a canceled operation, the API should return a consistent state that allows
+    continued service.
+
+-  A recommended minimum retention time for a stepwise operation is 24 hours.
+    Operations SHOULD transition to "tombstone" for an additional period of time
+    prior to being purged from the system.
+    
+-  Services that provides a new operation resource MUST support GET semantics on the operation.
+
+
+
+## Examples
+
+
+###  Create a new resource using RELO
+
+A client wants to provision a new database
+
+```
+POST https://graph.microsoft.com/v1.0/databases/
+
+{
+"id": "db1",
+}
+```
+
+The API responds synchronously that the database has been created and indicates
+that the provisioning operation is not fully completed by including the
+Operation-Location header and status property in the response payload.
+
+```
+HTTP/1.1 201 Created
+Content-Location: https://graph.microsoft.com/v1.0/databases/db1
+
+{
+"id": "db1",
+"Status": "Provisioning",
+[ … other fields for "database" …]
+}
+```
+
+### Create a new resource using the Stepwise Operation
+
+```
+POST https://graph.microsoft.com/v1.0/databases/
+
+{
+"id": "db1",
+}
+```
+
+The API responds synchronously that the request has been accepted and includes
+the Location header with an operation resource for further polling.
+
+```
+HTTP/1.1 202 Accepted
+
+Location: https://graph.microsoft.com/v1.0/operations/123
+
+```
+
+### Poll on a Stepwise Operation
+
+```
+
+GET https://graph.microsoft.com/v1.0/operations/123
+```
+
+The server responds that results are still not ready and optionally provides a
+recommendation to wait 30 seconds.
+
+```
+HTTP/1.1 200 OK
+Retry-After: 30
+
+{
+"createdDateTime": "2015-06-19T12-01-03.4Z",
+"status": "running"
+}
+```
+The client waits the recommended 30 seconds and then invokes another request to get
+the results of the operation.
+
+```
+GET https://graph.microsoft.com/v1.0/operations/123
+```
+
+
+The server responds with a "status:succeeded" operation that includes the resource
+location.
+
+```
+HTTP/1.1 200 OK
+
+{
+"createdDateTime": "2015-06-19T12-01-03.45Z",
+"lastActionDateTime": "2015-06-19T12-06-03.0024Z",
+"status": "succeeded",
+"resourceLocation": "https://graph.microsoft.com/v1.0/databases/db1"
+}
+```
```

Comment:
Do we want an example of how to cancel and operation for both resource-based and stepwise

Reply:
Added a new RELO cancelation example